packages/ui/src/components/shared/NotificationProvider.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Toaster } from 'react-hot-toast';
+
+type ToastType = 'success' | 'error' | 'loading' | 'blank' | 'custom';
+type ValueFunction<TValue, TArg> = (arg: TArg) => TValue;
+type ValueOrFunction<TValue, TArg> = TValue | ValueFunction<TValue, TArg>;
+type Renderable = JSX.Element | string | null;
+interface Toast {
+	type: ToastType;
+	id: string;
+	message: ValueOrFunction<Renderable, Toast>;
+	icon?: Renderable;
+	duration?: number;
+	pauseDuration: number;
+	position?: ToastPosition;
+	ariaProps: {
+		role: 'status' | 'alert';
+		'aria-live': 'assertive' | 'off' | 'polite';
+	};
+	style?: React.CSSProperties;
```

Comment:
Let's not allow defining custom styles. I would prefer keep using Tailwind as the only style library.

Reply:
Removed style prop